src/main/java/hudson/plugins/deploy/tomcat/TomcatAdapter.java

```diff
@@ -28,18 +33,24 @@
      * Top URL of Tomcat.
      */
     public final String url;
-    private final String path;
 
-    public TomcatAdapter(String url, String credentialsId) {
-        super(credentialsId);
-        this.url = url;
-        this.path = null;
-    }
+    /**
+     * Alternative context that override context defined in plugin main configuration
+     */
+    public final DeploymentContext alternativeDeploymentContext;
 
-    public TomcatAdapter(String url, String credentialsId, String path) {
+    private final String path;
+
+    public TomcatAdapter(
+        String url, String credentialsId, String alternativeDeploymentContext, String path) {
         super(credentialsId);
         this.url = url;
         this.path = path;
+        if(StringUtils.isNotBlank(alternativeDeploymentContext)) {
```

Comment:
Please use ternary if.

Reply:
I would like to use `StringUtils.defaultIfBlank`